pkg/antctl/raw/supportbundle/command_test.go

```diff
@@ -0,0 +1,363 @@
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+	fakerest "k8s.io/client-go/rest/fake"
+	cgtesting "k8s.io/client-go/testing"
+
+	"antrea.io/antrea/pkg/apis/crd/v1beta1"
+	systemv1beta1 "antrea.io/antrea/pkg/apis/system/v1beta1"
+	antrea "antrea.io/antrea/pkg/client/clientset/versioned"
+	fakeclientset "antrea.io/antrea/pkg/client/clientset/versioned/fake"
+	"antrea.io/antrea/pkg/client/clientset/versioned/scheme"
+)
+
+var (
+	clientConfig = &rest.Config{
+		APIPath: "/supportbundle",
+		ContentConfig: rest.ContentConfig{
+			NegotiatedSerializer: scheme.Codecs,
+			GroupVersion:         &systemv1beta1.SchemeGroupVersion,
+		},
+	}
+	controllerInfo = v1beta1.AntreaControllerInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "antrea-controller",
+		},
+		NodeRef: v1.ObjectReference{
+			Kind: "Node",
+			Name: "node-1",
+		},
+	}
+	node1 = v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "node-1",
+			ResourceVersion: "0",
+		},
+		Status: v1.NodeStatus{
+			Addresses: []v1.NodeAddress{
+				{
+					Type:    v1.NodeInternalIP,
+					Address: "172.16.0.0",
+				},
+			},
+		},
+	}
+	node2 = v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "node-1",
+		},
+	}
+	node3 = v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "node-3",
+			ResourceVersion: "0",
+		},
+		Status: v1.NodeStatus{
+			Addresses: []v1.NodeAddress{
+				{
+					Type:    v1.NodeInternalIP,
+					Address: "127.0.0.1",
+				},
+			},
+		},
+	}
+	agentInfo1 = &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "antrea-agent-1",
+		},
+		APIPort: 0,
+		PodRef: v1.ObjectReference{
+			ResourceVersion: "0",
+		},
+		NodeRef: v1.ObjectReference{
+			Kind: "Node",
+			Name: "node-1",
+		},
+	}
+	agentInfo2 = &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "antrea-agent-2",
+		},
+		APIPort: 0,
+		PodRef: v1.ObjectReference{
+			ResourceVersion: "0",
+		},
+		NodeRef: v1.ObjectReference{
+			Kind: "Node",
+			Name: "node-3",
+		},
+	}
+	nameList = []string{"node-1", "node-3"}
+)
+
+func TestLocalSupportBundleRequest(t *testing.T) {
+	getRestClient = func(cmd *cobra.Command) (rest.Interface, error) {
+		supportBundle := &systemv1beta1.SupportBundle{
+			ObjectMeta: metav1.ObjectMeta{Name: "agent"},
+			Status:     systemv1beta1.SupportBundleStatusCollected,
+		}
+		restClient := createFakeRESTClient(supportBundle)
+		return restClient, nil
+	}
+	defer func() {
+		getRestClient = setupRestClient
+	}()
+	cmd := &cobra.Command{}
+	cmd.SetContext(context.Background())
+	err := localSupportBundleRequest(cmd, "agent", os.Stdout)
```

Comment:
why the change here? originally you were using a buffer (instead of `os.Stdout`) and validating its contents after calling the function

Reply:
reverted back to original